pkg/ocm/managedcluster.go

```diff
@@ -0,0 +1,207 @@
+package ocm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	mcV1 "open-cluster-management.io/api/cluster/v1"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ManagedClusterBuilder provides struct for the ManagedCluster object containing connection to
+// the cluster and the ManagedCluster definitions.
+type ManagedClusterBuilder struct {
+	Definition *mcV1.ManagedCluster
+	Object     *mcV1.ManagedCluster
+	errorMsg   string
+	apiClient  runtimeClient.Client
+}
+
+// ManagedClusterAdditionalOptions additional options for ManagedCluster object.
+type ManagedClusterAdditionalOptions func(builder *ManagedClusterBuilder) (*ManagedClusterBuilder, error)
+
+// NewManagedClusterBuilder creates a new instance of ManagedClusterBuilder.
+func NewManagedClusterBuilder(apiClient *clients.Settings, name string) *ManagedClusterBuilder {
+	glog.V(100).Infof(
+		`Initializing new ManagedCluster structure with the following params: name: %s`, name)
+
+	builder := ManagedClusterBuilder{
+		apiClient: apiClient.Client,
+		Definition: &mcV1.ManagedCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: mcV1.ManagedClusterSpec{},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the ManagedCluster is empty")
+
+		builder.errorMsg = "ManagedCluster 'name' cannot be empty"
```

Comment:
Please use leading lowercase characters in your error messages
```suggestion
		builder.errorMsg = "managedcluster 'name' cannot be empty"
```

Reply:
Fixed

Reply:
I don't see the change yet